### ISSUE TITLE
java.lang.RuntimeException: Couldn't find content container view

### ISSUE THREAD

**Issue by tokunbo** [JakeWharton/ActionBarSherlock#1052]
I'm confused because I can't tell where the bug is. ActionBarSherlock or Robolectric; or what versions of stuff I should be using.

 Right now, I'm trying to use ABS4.4.0 & robolectric2.1.1. If I try using robolectric 2.3 from https://oss.sonatype.org/index.html#nexus-search;quick~org.robolectric , then I can't compile because ActionBarSherlockRobolectric.java from https://gist.github.com/JakeWharton/3803294 will not accept shadowOf(mActivity).setContentView(view); , setContentView is not a method of ShadowActivity in the newer robolectric.
 In Robolectric2.1.1, it compiles but I get the "Couldn't find content container view" anytime I try to create an activity: about_activity = Robolectric.buildActivity(AboutActivity.class).create().get();

What should I do? Does ABS4.4.0 work with Robolectric? If not, who fixes it? ....or does it even get fixed?

java.lang.RuntimeException: Couldn't find content container view
    at com.actionbarsherlock.internal.ActionBarSherlockCompat.generateLayout(ActionBarSherlockCompat.java:1015)
    at com.actionbarsherlock.internal.ActionBarSherlockCompat.installDecor(ActionBarSherlockCompat.java:902)
    at com.actionbarsherlock.internal.ActionBarSherlockCompat.initActionBar(ActionBarSherlockCompat.java:131)
    at com.actionbarsherlock.internal.ActionBarSherlockCompat.ensureActionBar(ActionBarSherlockCompat.java:1174)
    at com.actionbarsherlock.app.SherlockFragmentActivity.findViewById(SherlockFragmentActivity.java:280)
    at com.rackspace.cloudmobile.ui.AuthenticatorActivityTest.testClickLoginButton(AuthenticatorActivityTest.java:47)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:241)

**Comment by tokunbo** [comment]
So, blindly wandering about I decided to look at ActionBarSherlockCompat.java... I see that the following returns null:

```
ViewGroup contentParent = (ViewGroup)mDecor.findViewById(R.id.abs__content);
```

So, I tried changing that part of code into...

```
ViewGroup contentParent = (ViewGroup)mDecor.findViewById(R.id.abs__content);
if (contentParent == null) {
    contentParent = new LinearLayout(mActivity.getApplicationContext());
    //throw new RuntimeException("Couldn't find content container view");
    }
```

But that just created a nullptrEx somewhere else...

java.lang.NullPointerException
    at com.rackspace.cloudmobile.ui.AboutActivity.onCreate(AboutActivity.java:24)
    at android.app.Activity.performCreate(Activity.java:5008)
    at org.fest.reflect.method.Invoker.invoke(Invoker.java:112)
    at org.robolectric.util.ActivityController$1.run(ActivityController.java:119)
    at org.robolectric.shadows.ShadowLooper.runPaused(ShadowLooper.java:256)
    at org.robolectric.util.ActivityController.create(ActivityController.java:114)
    at org.robolectric.util.ActivityController.create(ActivityController.java:126)

**Comment by JakeWharton** [comment]
Robolectric 2.2 uses the real `PhoneWindow` for activities. That's going to be your best bet getting ABS working.

**Comment by tokunbo** [comment]
Hmm, ok. I've downloaded https://oss.sonatype.org/service/local/repositories/releases/content/org/robolectric/robolectric/2.2/robolectric-2.2.jar

So now I'm at the issue of "shadowOf(mActivity).setContentView(view);", where setContentView() is not defined for a ShadowActivity in ActionBarSherlockRobolectric.java . I'm kind-of new to AndroidDev; you mentioned "PhoneWindow", I assume you're talking about https://github.com/android/platform_frameworks_base/blob/master/policy/src/com/android/internal/policy/impl/PhoneWindow.java 

Does this mean I should try to replace the usage of Activity class in ActionBarSherlockRobolectric.java, to instead use PhoneWindow?

**Comment by JakeWharton** [comment]
You shouldn't need `ActionBarSherlockRobolectric` with 2.2. It uses a real window to hold activities. It should just work normally.

**Comment by tokunbo** [comment]
Oh really?!!! So I don't need this anymore?

```
ActionBarSherlock.registerImplementation(ActionBarSherlockRobolectric.class);
ActionBarSherlock.unregisterImplementation(ActionBarSherlockNative.class);
ActionBarSherlock.unregisterImplementation(ActionBarSherlockCompat.class);
```

Okay, I've commented that stuff out, now I got:

```
java.lang.RuntimeException: java.lang.IllegalArgumentException: Resource path must end in "/res"
```

I'm not sure what this is about, but I'll try and figure it out. I see a folder called "res" in my android project with things that look like resources.(anim  color  drawable  drawable-hdpi  drawable-ldpi  drawable-mdpi  drawable-tvdpi  drawable-xhdpi  drawable-xxhdpi  layout  layout-land  menu  values  values-v11  values-v14  xml) 

I'll see if I can get past this. Thanks a lot! 

EDIT:.... oh... https://github.com/JakeWharton/gradle-android-test-plugin/issues/28
Not entirely sure if this'll solve my issue, but trying stuff out...
